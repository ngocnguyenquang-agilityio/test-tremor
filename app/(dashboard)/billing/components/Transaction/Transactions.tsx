// Icons
import { FaRegCalendarAlt } from "react-icons/fa";

// Components
import TransactionList from "../TransactionList/TransactionList";

// Types
import { Transaction } from "@/types";

interface TransactionsProps {
  newest: Transaction[];
  date: string;
}

const Transactions = ({ newest, date }: TransactionsProps) => {
  return (
    <div className="p-6 bg-white dark:bg-[#202940] rounded-lg shadow-md">
      <div className="flex justify-between">
        <h6 className="font-semibold text-primary dark:text-white">
          Your Transaction&#39;s
        </h6>
        <div className="flex items-center text-secondary">
          <FaRegCalendarAlt className="dark:text-lighter" />
          <p className="text-sm dark:text-lighter ml-2">{date}</p>
        </div>
      </div>
      <h6 className="transaction-list-heading dark:text-lighter">NEWEST</h6>
      <div className="overflow-y-auto max-h-[505px]">
        <TransactionList transactions={newest} />
      </div>
    </div>
  );
};

export default Transactions;
