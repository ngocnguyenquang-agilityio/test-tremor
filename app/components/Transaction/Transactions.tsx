// Icons
import { FaCalendarAlt } from "react-icons/fa";

// Components
import { TransactionList } from "@/components";

// Types
import { Transaction } from "@/types";

interface TransactionsProps {
  newest: Transaction[];
  date: string;
}

const Transactions = ({ newest, date }: TransactionsProps) => {
  return (
    <div className="p-6 bg-white dark:bg-[#202940] rounded-lg shadow-md">
      <div className="flex justify-between">
        <h6 className="font-semibold text-[#344767] dark:text-white">
          Your Transactions
        </h6>
        <div className="flex items-center text-[#7b809a]">
          <FaCalendarAlt />
          <p className="text-sm ml-2">{date}</p>
        </div>
      </div>
      <h6 className="transaction-list-heading">NEWEST</h6>
      <div className="overflow-y-auto max-h-[520px]">
        <TransactionList transactions={newest} />
      </div>
    </div>
  );
};

export default Transactions;
