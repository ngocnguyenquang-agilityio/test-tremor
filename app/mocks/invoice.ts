export const INVOICE_DATA = {
  id: 230220,
  createdAt: "06/03/2019",
  dueDate: "11/03/2019",
  details: [
    {
      id: 1,
      name: "Premium Support",
      quantity: 1,
      rate: 9,
      amount: 89,
    },
    {
      id: 2,
      name: "Metror Dashboard",
      quantity: 3,
      rate: 100,
      amount: 90,
    },
    {
      id: 3,
      name: "Parts for service",
      quantity: 2,
      rate: 10,
      amount: 60,
    },
  ],
};
