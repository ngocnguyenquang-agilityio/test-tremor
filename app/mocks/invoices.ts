import { Invoice } from "@/types";

export const MOCK_INVOICES: Invoice[] = [
  {
    id: 804322,
    createdAt: "March, 21, 2021",
    invoicePrefix: "AR",
    totalCost: 180,
  },
  {
    id: 8908543,
    createdAt: "Apr, 22, 2022",
    invoicePrefix: "AR",
    totalCost: 180,
  },
  {
    id: 415645,
    createdAt: "May, 23, 2023",
    invoicePrefix: "MS",
    totalCost: 180,
  },
  {
    id: 103578,
    createdAt: "Jun, 24, 2024",
    invoicePrefix: "QW",
    totalCost: 180,
  },
  {
    id: 126749,
    createdAt: "Jul, 25, 2025",
    invoicePrefix: "RV",
    totalCost: 180,
  },
];
