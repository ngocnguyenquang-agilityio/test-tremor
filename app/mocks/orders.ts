// Types
import { Order } from "@/types";

export const MOCK_ORDERS: Order[] = [
  {
    id: 10425,
    createdAt: "2023-11-01T10:20:00+00:00",
    status: 1,
    customer: {
      id: 15,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 44.9,
    products: [
      {
        id: 6,
        name: "Phone Case Pink",
        count: 2,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 90,
      },
    ],
  },
  {
    id: 12345,
    createdAt: "2023-11-01T03:20:00+00:00",
    status: 0,
    customer: {
      id: 1,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 140.2,
    products: [
      {
        id: 12,
        name: "Office Papers",
        count: 1,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 77,
      },
    ],
  },
  {
    id: 12350,
    createdAt: "2023-11-01T11:03:00+00:00",
    status: 0,
    customer: {
      id: 9,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 112.5,
    products: [
      {
        id: 10,
        name: "Watter Bottle India",
        count: 3,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 7,
      },
    ],
  },
  {
    id: 12356,
    createdAt: "2023-11-01T09:20:00+00:00",
    status: 2,
    customer: {
      id: 13,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 23.9,
    products: [
      {
        id: 10,
        name: "Watter Bottle India",
        count: 3,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 34.35,
      },
    ],
  },
  {
    id: 12352,
    createdAt: "2023-11-01T16:05:00+00:00",
    status: 0,
    customer: {
      id: 16,
      avatar:
        "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/team-2.1593fb7f.jpg",
      fullName: "Orlando Imieto",
    },
    revenue: 350,
    products: [
      {
        id: 10,
        name: "Watter Bottle India",
        count: 3,
        url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
        price: 71,
      },
    ],
  },
];

export const ORDER_DETAIL_DATA = {
  id: 10421,
  created_at: "2023-11-01T03:20:00+00:00",
  order_code: "KF332",
  order_deliver_price: 14,
  order_tax: 1.95,
  track_order_info: {
    delivered_at: "2024-01-11T09:48:52Z",
    generate_order_at: "2024-01-11T01:50:52Z",
    generate_order_id: 1832412,
    transmited_to_courier_at: "2024-01-11T02:48:52Z",
  },
  billing_info: {
    vat: "FB1235476",
    email: "olover@burrito.com",
    last4_bank: "1234",
    owner_name: "Oliver Liam",
    company_name: "Viking@burrito.com",
  },
};
