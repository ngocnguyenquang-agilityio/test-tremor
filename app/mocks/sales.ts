export const STATISTICS_DATA = [
  {
    id: "1",
    type: "Sales",
    amount: "$230,220",
    totalAmount: "55%",
    duration: "since last month",
  },
  {
    id: "2",
    type: "Customers",
    amount: "3.200",
    totalAmount: "12%",
    duration: "since last month",
  },
  {
    id: "3",
    type: "Avg. Revenue",
    amount: "$1.200",
    totalAmount: "$213",
    duration: "since last month",
  },
];

export const TOP_SELLING_PRODUCTS_DATA = [
  {
    product_id: 1,
    name: "Nike v22 Running",
    orders: "8.232",
    value: "$130.992",
    ads_spent: "$9.500",
    refunds: 13,
    refunds_type: "increase",
  },
  {
    product_id: 2,
    name: "Business Kit (Mug + Notebook)",
    orders: "12.821",
    value: "$80.250",
    ads_spent: "$4.200",
    refunds: 40,
    refunds_type: "decrease",
  },
  {
    product_id: 3,
    name: "Black Chair",
    orders: "2.421",
    value: "$40.600",
    ads_spent: "$9.430",
    refunds: 54,
    refunds_type: "increase",
  },
  {
    product_id: 4,
    name: "Wireless Charger",
    orders: "5.921",
    value: "$91.300",
    ads_spent: "$7.364",
    refunds: 5,
    refunds_type: "decrease",
  },
  {
    product_id: 5,
    name: "Mountain Trip Kit (Camera + Backpack)",
    orders: "921",
    value: "$140.925",
    ads_spent: "$20.531",
    refunds: 121,
    refunds_type: "increase",
  },
];

export const REVENUE_CHART_DATA = {
  title: "Revenue",
  data: [
    {
      date: "Apr",
      "Facebook Ads": 50,
      "Google Ads": 0,
    },
    {
      date: "May",
      "Facebook Ads": 100,
      "Google Ads": 30,
    },
    {
      date: "Jun",
      "Facebook Ads": 200,
      "Google Ads": 40,
    },
    {
      date: "Jul",
      "Facebook Ads": 200,
      "Google Ads": 110,
    },
    {
      date: "Aug",
      "Facebook Ads": 400,
      "Google Ads": 150,
    },
    {
      date: "Sep",
      "Facebook Ads": 340,
      "Google Ads": 210,
    },
    {
      date: "Oct",
      "Facebook Ads": 500,
      "Google Ads": 290,
    },
    {
      date: "Nov",
      "Facebook Ads": 450,
      "Google Ads": 250,
    },
    {
      date: "Dec",
      "Facebook Ads": 700,
      "Google Ads": 290,
    },
  ],
};
export const SALES_AGE_CHART = [
  {
    value: 34,
    ageRange: "16-20",
  },
  {
    value: 23,
    ageRange: "21-25",
  },
  {
    value: 45,
    ageRange: "26-30",
  },
  {
    value: 53,
    ageRange: "36-40",
  },
  {
    value: 26,
    ageRange: "40+",
  },
];
