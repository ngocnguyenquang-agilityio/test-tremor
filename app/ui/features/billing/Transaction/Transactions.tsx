// Icons
import { FaRegCalendarAlt } from "react-icons/fa";

// Components
import { Text } from "@tremor/react";
import TransactionList from "../TransactionList/TransactionList";

// Types
import { Transaction } from "@/types";
import CreateTransactionContainer from "./CreateTransactionContainer";

interface TransactionsProps {
  newest: Transaction[];
  date: string;
}

const Transactions = ({ newest, date }: TransactionsProps) => {
  return (
    <div className="p-6 bg-white dark:bg-[#202940] rounded-lg shadow-md">
      <div className="flex justify-between">
        <div className="flex gap-4">
          <h6 className="font-semibold text-primary dark:text-white">
            Your Transaction&#39;s
          </h6>
          <CreateTransactionContainer />
        </div>
        <div className="flex items-center text-secondary">
          <FaRegCalendarAlt className="dark:text-lighter" />
          <Text className="dark:text-lighter ml-2">{date}</Text>
        </div>
      </div>
      <h6 className="transaction-list-heading dark:text-lighter">NEWEST</h6>
      <div className="overflow-y-auto max-h-[505px]">
        <TransactionList transactions={newest} />
      </div>
    </div>
  );
};

export default Transactions;
