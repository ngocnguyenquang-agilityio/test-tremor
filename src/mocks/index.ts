export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
