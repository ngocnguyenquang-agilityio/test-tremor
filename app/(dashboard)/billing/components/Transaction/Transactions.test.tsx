// Libs
import { render } from "@testing-library/react";

// Components
import Transactions from "./Transactions";

// Mocks
import { MOCK_TRANSACTIONS } from "@/mocks/transaction";

describe("Transactions Component Testing", () => {
  const newest = MOCK_TRANSACTIONS;

  it("should match snapshot", () => {
    const { container } = render(
      <Transactions newest={newest} date="23 - 30 March 2020" />,
    );

    expect(container).toMatchSnapshot();
  });
});
