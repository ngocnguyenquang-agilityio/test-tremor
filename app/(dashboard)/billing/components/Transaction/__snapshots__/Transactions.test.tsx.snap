// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions Component Testing should match snapshot 1`] = `
<div>
  <div
    class="p-6 bg-white dark:bg-[#202940] rounded-lg shadow-md"
  >
    <div
      class="flex justify-between"
    >
      <h6
        class="font-semibold text-primary dark:text-white"
      >
        Your Transaction's
      </h6>
      <div
        class="flex items-center text-secondary"
      >
        <svg
          class="dark:text-lighter"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
          />
        </svg>
        <p
          class="text-sm dark:text-lighter ml-2"
        >
          23 - 30 March 2020
        </p>
      </div>
    </div>
    <h6
      class="transaction-list-heading dark:text-lighter"
    >
      NEWEST
    </h6>
    <div
      class="overflow-y-auto max-h-[505px]"
    >
      <ul>
        <div
          class="flex pr-4 py-4 justify-between items-center"
        >
          <div
            class="flex gap-4 items-center"
          >
            <button
              class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-attention border-attention"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                />
              </svg>
            </button>
            <div>
              <h6
                class="text-sm font-semibold text-primary dark:text-white mb-1"
              >
                Netflix
              </h6>
              <p
                class="text-xs text-secondary dark:text-lighter"
              >
                05 Feb 2024, at 09:50 AM
              </p>
            </div>
          </div>
          <p
            class="text-sm font-semibold text-attention"
          >
            - $2,500.00
          </p>
        </div>
        <div
          class="flex pr-4 py-4 justify-between items-center"
        >
          <div
            class="flex gap-4 items-center"
          >
            <button
              class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-few border-few"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
                />
              </svg>
            </button>
            <div>
              <h6
                class="text-sm font-semibold text-primary dark:text-white mb-1"
              >
                Apple
              </h6>
              <p
                class="text-xs text-secondary dark:text-lighter"
              >
                05 Feb 2024, at 09:50 AM
              </p>
            </div>
          </div>
          <p
            class="text-sm font-semibold text-few"
          >
            + $2,000.00
          </p>
        </div>
        <div
          class="flex pr-4 py-4 justify-between items-center"
        >
          <div
            class="flex gap-4 items-center"
          >
            <button
              class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-few border-few"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
                />
              </svg>
            </button>
            <div>
              <h6
                class="text-sm font-semibold text-primary dark:text-white mb-1"
              >
                Stripe
              </h6>
              <p
                class="text-xs text-secondary dark:text-lighter"
              >
                05 Feb 2024, at 09:51 AM
              </p>
            </div>
          </div>
          <p
            class="text-sm font-semibold text-few"
          >
            + $750.00
          </p>
        </div>
        <div
          class="flex pr-4 py-4 justify-between items-center"
        >
          <div
            class="flex gap-4 items-center"
          >
            <button
              class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-few border-few"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
                />
              </svg>
            </button>
            <div>
              <h6
                class="text-sm font-semibold text-primary dark:text-white mb-1"
              >
                HubSpot
              </h6>
              <p
                class="text-xs text-secondary dark:text-lighter"
              >
                05 Jan 2024, at 09:51 AM
              </p>
            </div>
          </div>
          <p
            class="text-sm font-semibold text-few"
          >
            + $1,000.00
          </p>
        </div>
        <div
          class="flex pr-4 py-4 justify-between items-center"
        >
          <div
            class="flex gap-4 items-center"
          >
            <button
              class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-few border-few"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
                />
              </svg>
            </button>
            <div>
              <h6
                class="text-sm font-semibold text-primary dark:text-white mb-1"
              >
                Creative Tim
              </h6>
              <p
                class="text-xs text-secondary dark:text-lighter"
              >
                05 Feb 2024, at 09:52 AM
              </p>
            </div>
          </div>
          <p
            class="text-sm font-semibold text-few"
          >
            + $2,500.00
          </p>
        </div>
        <div
          class="flex pr-4 py-4 justify-between items-center"
        >
          <div
            class="flex gap-4 items-center"
          >
            <button
              class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-primary border-primary"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.005 3.1a1 1 0 1 1 1.99 0l-.388 6.35a.61.61 0 0 1-1.214 0L7.005 3.1ZM7 12a1 1 0 1 1 2 0 1 1 0 0 1-2 0Z"
                />
              </svg>
            </button>
            <div>
              <h6
                class="text-sm font-semibold text-primary dark:text-white mb-1"
              >
                Webflow
              </h6>
              <p
                class="text-xs text-secondary dark:text-lighter"
              >
                05 Feb 2024, at 09:52 AM
              </p>
            </div>
          </div>
          <p
            class="text-sm font-semibold text-primary"
          >
            Pending
          </p>
        </div>
        <div
          class="flex pr-4 py-4 justify-between items-center"
        >
          <div
            class="flex gap-4 items-center"
          >
            <button
              class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-few border-few"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
                />
              </svg>
            </button>
            <div>
              <h6
                class="text-sm font-semibold text-primary dark:text-white mb-1"
              >
                Momo
              </h6>
              <p
                class="text-xs text-secondary dark:text-lighter"
              >
                05 Feb 2024, at 09:52 AM
              </p>
            </div>
          </div>
          <p
            class="text-sm font-semibold text-few"
          >
            + $90.00
          </p>
        </div>
      </ul>
    </div>
  </div>
</div>
`;
